### ISSUE TITLE
Something wrong with the Swagger specs on developer.couchbase.com?

### ISSUE THREAD

**Issue by Prudwell** [couchbaselabs/couchbase-mobile-portal#520]
I'm using the swagger specs hosted on developer.couchbase.com for development and this morning I'm getting an error

```
Possible Unhandled Promise Rejection (id: 14):
Cannot read property 'post_db_session' of undefined
```

And when I go check the swagger UI page there's an error so I can't check if the method name I'm trying to use has changed.

[https://developer.couchbase.com/mobile/swagger/sync-gateway-public/](https://developer.couchbase.com/mobile/swagger/sync-gateway-public/)

"Can't read from server. It may not have the appropriate access-control-origin settings."

<img width="870" alt="screen shot 2016-12-27 at 10 48 50 am" src="https://cloud.githubusercontent.com/assets/11758003/21506083/43cf3c8e-cc22-11e6-85bb-e56d48117db5.png">

The direct link to the spec still seems to work

[https://developer.couchbase.com/mobile/swagger/sync-gateway-public/spec.json](https://developer.couchbase.com/mobile/swagger/sync-gateway-public/spec.json) 

Anyone else getting an issue like this? I can't seem to get my code to output any useful caught error or log of what's going wrong.

I'll update if I discover more

**Comment by Prudwell** [comment]
Ah, turned out to be related to my laptop connecting to the internet through a VPN. After turning that off my app is working, though the the swagger ui page at 
https://developer.couchbase.com/mobile/swagger/sync-gateway-public/ still is not showing the UI of the spec.

Ok note to future self, make sure VPN is off before starting development.

**Comment by jamiltz** [comment]
Thanks, swagger ui is now embedded [in the docs](https://developer.couchbase.com/documentation/mobile/current/references/couchbase-lite/rest-api/index.html). I've requested redirects from the standalone links to the embedded ones.

I'm curious to know, what are you using the spec(s) for? (web dev, mobile, app server..?)

**Comment by Prudwell** [comment]
Oh gotcha I see it now.

And for your question: all of the above. React Native app uses the public endpoint for creating sessions, the app server does lots of talking to both public and admin endpoints. The web app side of things is barely started but I have sessions and some basic functionality working using the public endpoint with swagger client in the browser.

I see now that react-native-couchbase-lite is using the swagger spec for CBL. I'm using that package but the older pre-swagger version. Will look to updating my code to use the new version soon, hoping it's not too painful. It will be nice to have swagger everywhere once I have that done.

Hey I've seen talk of changelogs for the swagger specs in the past, is that something that's happened? There were a few instances where I was using the hosted specs and something changed in the spec that I had to track down manually to get my code working again. That hasn't happened in a long time though so maybe the specs are stable now.

Cheers

**Comment by jamiltz** [comment]
Great to hear that you are using the swagger specs with swagger clients extensively!

> I see now that react-native-couchbase-lite is using the swagger spec for CBL. I'm using that package but the older pre-swagger version. Will look to updating my code to use the new version soon, hoping it's not too painful. It will be nice to have swagger everywhere once I have that done.

Yes that's the plan. It's on master but we've not published it to npm yet. Waiting to have the example app [completed](https://github.com/couchbaselabs/react-native-couchbase-lite/pull/108).

> Hey I've seen talk of changelogs for the swagger specs in the past, is that something that's happened? There were a few instances where I was using the hosted specs and something changed in the spec that I had to track down manually to get my code working again. That hasn't happened in a long time though so maybe the specs are stable now.

Not yet, we've been busy working on the [tutorial content](https://developer.couchbase.com/documentation/mobile/1.3/training/index.html) for CBM which the react native sample app will be part of. The change log might be something we look at after that. It needs to be done when we build the spec and maybe use the compare view on github to display a diff of the specs at two different commits.
